File: Arcade/TheCore/Kotlin/candles.kt
fun candles(candlesNumber: Int, makeNew: Int): Int {
    var leftovers: Int = 0;
    var totalCandles: Int = 0;
    var candlesToBurn: Int = candlesNumber;
    
    while (candlesToBurn >= 1) {
        candlesToBurn--;
        leftovers++;
        totalCandles++;
        
        if (leftovers == makeNew) {
            candlesToBurn++;
            leftovers = 0;
        }
    }
    
    return totalCandles;
}

File: Arcade/TheCore/Javascript/candles.js
function candles(candlesNumber, makeNew) {
    let leftovers = 0;
    let totalCandles = 0;
    
    while (candlesNumber >= 1) {
        candlesNumber--;
        leftovers++
        totalCandles++;
        
        if (leftovers == makeNew) {
            candlesNumber++;
            leftovers = 0;
        }
    }
    
    return totalCandles;
}